Ganeti 3.1.1 support

- https://github.com/ganeti/ganeti/releases/tag/v3.1.1

```bash
# make py-tests-legacy
...
Running ganeti.hooks_unittest.py
......F........
======================================================================
FAIL: testEnv (__main__.TestHooksRunner.testEnv)
Test environment execution
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/gntbuild.DDQwg1LN/test/py/legacy/ganeti.hooks_unittest.py", line 208, in testEnv
    self.assertEqual(self.hr.RunHooks(self.hpath, phase, env_snt),
AssertionError: Lists differ: [('fake-pre.d/success', 2, 'PHASE=pre\\nSHLVL=1\\n_=/usr/bin/env')] != [('fake-pre.d/success', 2, 'PHASE=pre')]

First differing element 0:
('fake-pre.d/success', 2, 'PHASE=pre\\nSHLVL=1\\n_=/usr/bin/env')
('fake-pre.d/success', 2, 'PHASE=pre')

- [('fake-pre.d/success', 2, 'PHASE=pre\\nSHLVL=1\\n_=/usr/bin/env')]
+ [('fake-pre.d/success', 2, 'PHASE=pre')]

----------------------------------------------------------------------
Ran 15 tests in 0.022s

FAILED (failures=1)
...
```

--- a/.github/workflows/release-drafter.yml
+++ b/.github/workflows/release-drafter.yml
@@ -4,12 +4,6 @@ on:
   push:
     branches:
       - main
-  pull_request:
-    types:
-      - opened
-      - reopened
-      - edited
-      - synchronize
 
 permissions:
   contents: read

--- a/.github/workflows/release-drafter.yml
+++ b/.github/workflows/release-drafter.yml
@@ -22,6 +22,6 @@ jobs:
       pull-requests: write
 
     steps:
-      - uses: release-drafter/release-drafter@b1476f6e6eb133afa41ed8589daba6dc69b4d3f5 # v6.1.0
+      - uses: release-drafter/release-drafter@5de93583980a40bd78603b6dfdcda5b4df377b32 # v7.2.0
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -29,7 +29,7 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
 
       - name: Build RPM
         id: build_rpm

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -31,7 +31,7 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
 
       - name: Build RPM
         id: build_rpm
@@ -39,7 +39,7 @@ jobs:
           ./build $(echo ${BUILD_IMAGE} | sed 's/-/:/g') -a -bi
 
       - name: Upload artifacts
-        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+        uses: actions/upload-artifact@043fb46d1a93c77aae656e7c1c64a875d1fc6a0a # v7.0.1
         with:
           name: ${{ matrix.image }}
           path: rpmbuild/**/*.rpm

--- a/README.md
+++ b/README.md
@@ -6,9 +6,9 @@ Ganeti RPM Packaging for RHEL/AlmaLinux/Rocky Linux/others.
 
 ## Packaging status
 
-- RHEL/AlmaLinux/Rocky Linux/others 10.x: 3.1.0-1
-- RHEL/AlmaLinux/Rocky Linux/others 9.x: 3.1.0-1
-- RHEL/AlmaLinux/Rocky Linux/others 8.x: 3.1.0-1
+- RHEL/AlmaLinux/Rocky Linux/others 10.x: 3.1.1-1
+- RHEL/AlmaLinux/Rocky Linux/others 9.x: 3.1.1-1
+- RHEL/AlmaLinux/Rocky Linux/others 8.x: 3.1.1-1
 
 ## Documentation
 

--- a/rpmbuild/ganeti/SOURCES/ganeti-3.1.1-fix-legacy-unittest.patch
+++ b/rpmbuild/ganeti/SOURCES/ganeti-3.1.1-fix-legacy-unittest.patch
@@ -1,0 +1,19 @@
+diff -urN ganeti-3.1.1.default/test/py/legacy/ganeti.hooks_unittest.py ganeti-3.1.1/test/py/legacy/ganeti.hooks_unittest.py
+--- ganeti-3.1.1.default/test/py/legacy/ganeti.hooks_unittest.py	2026-04-27 15:10:44.000000000 +0000
++++ ganeti-3.1.1/test/py/legacy/ganeti.hooks_unittest.py	2026-04-28 01:01:16.329888987 +0000
+@@ -204,9 +204,12 @@
+       os.chmod(fname, 0o755)
+       self.torm.append((fname, False))
+       env_snt = {"PHASE": phase}
+-      env_exp = "PHASE=%s" % phase
+-      self.assertEqual(self.hr.RunHooks(self.hpath, phase, env_snt),
+-                           [(self._rname(fname), HKR_SUCCESS, env_exp)])
++      env_exp = "PHASE=%s\\nSHLVL=1\\n_=/usr/bin/env" % phase
++      result = self.hr.RunHooks(self.hpath, phase, env_snt)
++      self.assertEqual(len(result), 1)
++      self.assertEqual(result[0][0], self._rname(fname))
++      self.assertEqual(result[0][1], HKR_SUCCESS)
++      self.assertEqual(env_exp, result[0][2])
+ 
+ 
+ def FakeHooksRpcSuccess(node_list, hpath, phase, env):

--- a/rpmbuild/ganeti/SPECS/ganeti.spec
+++ b/rpmbuild/ganeti/SPECS/ganeti.spec
@@ -27,7 +27,7 @@
 %define extstorage_search_path %{_search_sharedir}/%{name}/extstorage,%{_search_libdir}/%{name}/extstorage,%{_search_lib64dir}/%{name}/extstorage,%{_search_local_libdir}/%{name}/extstorage,%{_search_local_lib64dir}/%{name}/extstorage,/srv/%{name}/extstorage
 
 Name: ganeti
-Version: 3.1.0
+Version: 3.1.1
 Release: 1%{?dist}
 Group: System Environment/Daemons
 Summary: Cluster virtual server management software
@@ -53,6 +53,7 @@ Patch11: ganeti-3.1.0-kvm-qmp-timeout.patch
 Patch12: ganeti-3.1.0-ignore-test-start-daemon.patch
 # ignore tests with AssertionError: ResolverError not raised by <lambda>
 Patch13: ganeti-3.1.0-ignore-test-hostname-resolution-error.patch
+Patch14: ganeti-3.1.1-fix-legacy-unittest.patch
 
 BuildRequires: iproute
 BuildRequires: libcurl-devel
@@ -148,6 +149,7 @@ It is not required when the init system used is systemd.
 %patch -P11 -p1
 %patch -P12 -p1
 %patch -P13 -p1
+%patch -P14 -p1
 
 %build
 %if 0%{?rhel} == 10
@@ -329,6 +331,10 @@ usermod -aG gnt-daemons gnt-rapi
 %config(noreplace) %{_sysconfdir}/sysconfig/%{name}
 
 %changelog
+* Tue Apr 28 2026 Jun Futagawa <jfut@integ.jp> - 3.1.1-1
+- Update to 3.1.1 (#74)
+- Add ganeti-3.1.1-fix-legacy-unittest.patch (#74)
+
 * Thu Sep 25 2025 Jun Futagawa <jfut@integ.jp> - 3.1.0-1
 - Update to 3.1.0 (#53)
 - Drop support for RHEL/CentOS 7

